### PR TITLE
feat(spectacular)!: add required `routes` option and optional testing features for `provideSpectacularFeatureTest`

### PR DESCRIPTION
**Pull request by LayZeeDK**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[x] Documentation content changes
[ ] Other... Please describe:
```

### Features
1. `provideSpectacularFeatureTest` only supports the `featurePath` option
1. `provideSpectacularFeatureTest` returns `Provider[]`
1. `provideSpectacularFeatureTest` is stable and `provideSpectacularFeatureTesting` does not exist
1. `withInitialFeatureNavigation` does not exist
1. `SpectacularFeatureTestingModule` is stable

### Refactors
1. `SpectacularFeatureTestingRootModule` uses `InjectFlags`

### Testing
1. A classic `CapitalizePipe` is tested

### Documentation
1. The `excludeComponentDeclaration` option is specified for all Angular Testing Library example tests

Issue Number: N/A

### Features
1. `provideSpectacularFeatureTest` requires a `routes` option and optionally accepts the `withInitialFeatureNavigation` feature and Angular Router features
1. `provideSpectacularFeatureTest` returns `(EnvironmentProviders | Provider)[]`
1. `provideSpectacularFeatureTest` is marked as deprecated and a copy is renamed to `provideSpectacularFeatureTesting`
1. Add `withInitialFeatureNavigation` for use with `provideSpectacularFeatureTest` and `provideSpectacularFeatureTesting`
1. `SpectacularFeatureTestingModule` is deprecated

### Refactors
1. `SpectacularFeatureTestingRootModule` uses `InjectOptions`

### Testing
1. A classic and a standalone `CapitalizePipe` are tested

### Documentation
1. The `excludeComponentDeclaration` option is not mentioned in Angular Testing Library example tests

## Does this PR introduce a breaking change?

```
[x] Yes
[ ] No
```

### BREAKING CHANGES
- A `routes` option must be passed to `provideSpectacularFeatureTest`.
- `provideSpectacularFeatureTest` returns `(EnvironmentProviders | Provider)[]`

### DEPRECATIONS
- `provideSpectacularFeatureTest` is to be removed in Spectacular version 16. Migrate to `provideSpectacularFeatureTesting`.
- `SpectacularFeatureTestingModule` is to be removed in Spectacular version 16. Migrate to `provideSpectacularFeatureTesting`.

## Other information
[API design Gist](https://gist.github.com/LayZeeDK/0ec8fd8f3499f019bf85c07f5a3259fd)

### CODE REVIEW

File: packages/spectacular/src/lib/feature-testing/configuration/spectacular-feature-testing-features.ts [hunk 1]
```diff
@@ -0,0 +1,14 @@
+import type { Provider } from '@angular/core';
+
+export type SpectacularFeatureTestingFeatureKind = 'InitialFeatureNavigation';
+
+export interface SpectacularFeatureTestingFeature<
+  TFeatureKind extends SpectacularFeatureTestingFeatureKind
+> {
+  readonly ɵbrand: 'SpectacularFeatureTestingFeature';
+  readonly ɵkind: TFeatureKind;
+  readonly ɵproviders: Provider[];
```
Comment (NachoVazquez) [inline packages/spectacular/src/lib/feature-testing/configuration/spectacular-feature-testing-features.ts:10]: question: I remember that you mentioned that the `ɵ` was unstable and prone to changes. Is this different?

**Comment by NachoVazquez** [comment]
It is looking great; I'm excited.

File: packages/spectacular/src/lib/feature-testing/configuration/spectacular-feature-testing-features.ts (hunk 1 above)
Comment (LayZeeDK) [reply packages/spectacular/src/lib/feature-testing/configuration/spectacular-feature-testing-features.ts:10]: In Angular packages, it means *internal* but can also mean experimental/unstable. Here, I'm following the pattern of Angular HTTP and Angular Router tree-shakable feature types to support passing Angular Router features to `provideSpectacularFeatureTesting`. I'm going to document these types before this PR is merged.

Comment (LayZeeDK) [reply packages/spectacular/src/lib/feature-testing/configuration/spectacular-feature-testing-features.ts]: These properties have been marked as `@internal` so they won't show up in the API reference and this annotation will be visible to library consumers navigating the type definitions.

File: packages/spectacular/src/lib/feature-testing/configuration/spectacular-feature-testing-features.ts [hunk 2]
```diff
@@ -0,0 +1,53 @@
+import type { Provider } from '@angular/core';
+import type { provideSpectacularFeatureTesting } from './provide-spectacular-feature-testing';
```
Comment (NachoVazquez) [inline packages/spectacular/src/lib/feature-testing/configuration/spectacular-feature-testing-features.ts:2]: nitpick: nice warning from the linter that `provideSpectacularFeatureTesting` is not being used

File: packages/spectacular/src/lib/feature-testing/configuration/with-initial-feature-navigation.ts [hunk 3]
```diff
@@ -0,0 +1,43 @@
+import { initialFeatureNavigationInitializer } from '../navigation/initial-feature-navigation.initializer';
+import { makeSpectacularFeatureTestingFeature } from './make-spectacular-feature-testing-feature';
+import type { provideSpectacularFeatureTesting } from './provide-spectacular-feature-testing';
```
Comment (NachoVazquez) [inline packages/spectacular/src/lib/feature-testing/configuration/with-initial-feature-navigation.ts:3]: nitpick: a nice warning from the linter that provideSpectacularFeatureTesting is not being used

**Review by NachoVazquez — CHANGES_REQUESTED** [review]
I left some small nitpicks assisted by the linter, but it looks great!

File: packages/spectacular/src/lib/feature-testing/configuration/with-initial-feature-navigation.ts (hunk 3 above)
Comment (LayZeeDK) [reply packages/spectacular/src/lib/feature-testing/configuration/with-initial-feature-navigation.ts:3]: It's used by TSDoc `@link` annotations.

File: packages/spectacular/src/lib/feature-testing/configuration/spectacular-feature-testing-features.ts (hunk 2 above)
Comment (LayZeeDK) [reply packages/spectacular/src/lib/feature-testing/configuration/spectacular-feature-testing-features.ts:2]: It's used by TSDoc `@link` annotations.

**Pull request merged**